File: migrate_csvs_to_blob.py
import os, requests
from pathlib import Path

BLOB_TOKEN = "your_BLOB_READ_WRITE_TOKEN_here"
BASE_URL = "https://blob.vercel-storage.com"

for csv_file in Path("data/csv").glob("*.csv"):
    with open(csv_file, "rb") as f:
        content = f.read()
    r = requests.put(
        f"{BASE_URL}/csv/{csv_file.name}",
        headers={
            "Authorization": f"Bearer {BLOB_TOKEN}",
            "Content-Type": "text/csv",
            "x-content-type": "text/csv",
        },
        data=content
    )
    print(f"{'✅' if r.status_code in (200, 201) else '❌'} {csv_file.name} → HTTP {r.status_code}")

File: migrate_csvs_to_blob.ts
import { put } from "@vercel/blob";
import { readdirSync, readFileSync } from "fs";
import { join } from "path";

const token = process.env.BLOB_READ_WRITE_TOKEN!;
const csvDir = join(process.cwd(), "data", "csv");

const files = readdirSync(csvDir).filter(f => f.endsWith(".csv"));
console.log(`Found ${files.length} CSV files to upload...`);

for (const filename of files) {
  const content = readFileSync(join(csvDir, filename));
  const blob = await put(`csv/${filename}`, content, {
    access: "public",
    token,
    addRandomSuffix: false,
    contentType: "text/csv",
  });
  console.log(`✅ ${filename} → ${blob.url}`);
}
